core/base_algo.py

```diff
@@ -371,26 +372,31 @@ def get_remaining_categories(self) -> Set:
         return ALL_CATEGORIES - categories_in_file
 
     def _update_results_csv(self, values_dict) -> None:
-        if self.args.category in set(self.experiment_results_manager.data.category):
-            mask = (self.experiment_results_manager.data.category != self.args.category)
+        key = [self.args.category, self.model_name]
+        if self.experiment_results_manager.data[['category', 'model_name']].isin(key).any(axis=1).any():
+            # Key match found, overwriting
+            mask = (self.experiment_results_manager.data[['category', 'model_name']] != key)
             self.experiment_results_manager.data = self.experiment_results_manager.data[mask]
         
         new_dict = values_dict.copy()
         new_dict['category'] = self.args.category
         new_dict['category_type'] = CATEGORY_TO_TYPE[self.args.category]
+        new_dict['model_name'] = self.model_name
```

Comment:
this should be the first assignment to the new_dict imo.

Reply:
I'm keeping it as the third entry so this is also the third assignment